### ISSUE TITLE
Unexpected super classes cause ClassCastException

### ISSUE THREAD

**Issue by pwalch** [devsearch-epfl/devsearch-ast#12]
ClassDefExtractor fails when an unexpected super class is found, at this [location](https://github.com/devsearch-epfl/devsearch-ast/blob/82b33a95f3f6fbee4a60c787203bad656fcd7b8f/src/main/scala/devsearch/features/ClassDefExtractor.scala#L17).

Top of stack trace:
```
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 26 in stage 0.0 failed 4 times, most recent failure: Lost task 26.3 in stage 0.0 (TID 63, icdatasrv008.icdatacluster2): java.lang.ClassCastException: devsearch.ast.PrimitiveTypes$String$ cannot be cast to devsearch.ast.ClassType
    at devsearch.features.ClassDefExtractor$$anonfun$extract$1$$anonfun$apply$1.isDefinedAt(ClassDefExtractor.scala:17)
```

After some investigation, I found a file to reproduce:
* apache/axis2-java/modules/metadata/src/org/apache/axis2/jaxws/description/xml/handler/EjbLinkType.java

```
/*
license
 */

//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference
//


package org.apache.axis2.jaxws.description.xml.handler;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlType;


/** doc */
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "ejb-linkType")
public class EjbLinkType
        extends String {


}
```

Here is another file that throws the exception:
* apache/axis2-java/modules/metadata/src/org/apache/axis2/jaxws/description/xml/handler/PathType.java

Those two files contain a class extending String, which is quite unexpected and wrong :smile: 
I added a quick fix [here](https://github.com/devsearch-epfl/devsearch-ast/commit/25fe71475c697ee05458e4dd98dd002c60939fef). @samarion , do you know a better way?

**Comment by samarion** [comment]
Well this is very weird! List.collect takes a partial function that is simply not defined for PrimitiveTypes.String so the code should actually be correct. Seems like a bug in the compiler...

I guess a nicer fix would be to go through flatMap where case ClassType returns List(feature) and case _ returns Nil, but catching bugs with a try-catch is fine. There's no nice fix for compiler bugs, only work-arounds.

You can merge your fix.

**Comment by pwalch** [comment]
Fixed by #13
